### PR TITLE
Add cookbook and meta example for Canberra Distance

### PR DESCRIPTION
**Pull request by syashakash**
Issues #3000, #3555, #4047 
All tests passed. No memory leaks found yet.

### CODE REVIEW

File: doc/cookbook/source/examples/distance/canberra.rst [hunk 1]
```diff
@@ -0,0 +1,39 @@
+==================
+Canberra Distance
+==================
+
+The Canberra distance between vectors P and Q is the similarity as measured by their angle.
```
Comment (vigsterkr) [inline doc/cookbook/source/examples/distance/canberra.rst]: by their angle?
it's a weighted manhattan distance...

File: doc/cookbook/source/examples/distance/canberra.rst [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+-------
+Example
+-------
+
+Imagine we have files with data. We create CDenseFeatures (here 64 bit floats aka RealFeatures) as
+
+.. sgexample:: canberra.sg:create_features
+
+We create an instance of :sgclass:`CCanberraMetric` by passing it :sgclass:`CDenseFeatures`.
```
Comment (vigsterkr) [inline doc/cookbook/source/examples/distance/canberra.rst]: you mean something like:
We can create an instance of CanberraMetric by passing two Features objects for which features  we would like to measure the pairwise Canberra distances.

File: doc/cookbook/source/examples/distance/canberra.rst (hunk 1 above)
Comment (syashakash) [reply doc/cookbook/source/examples/distance/canberra.rst]: Corrected.

**Comment by karlnapf** [comment]
I wrote in #3000 
"No need for cookbook rst pages for the remaining distance examples. A meta example is fine."
But ok, let's have one more :)
Thanks for the patch!

File: examples/meta/src/distance/canberra.sg [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+CanberraMetric distance(features_a, features_a)
+#![create_instance]
+
+#![extract_distance]
+RealMatrix distance_matrix_aa = distance.get_distance_matrix()
+#![extract_distance]
+
+#![refresh_distance]
+distance.init(features_a, features_b)
+RealMatrix distance_matrix_ab = distance.get_distance_matrix()
```
Comment (karlnapf) [inline examples/meta/src/distance/canberra.sg]: could you send a PR to `data` with the integration testing data for this example? And then update this PR with that new `data` revision?

Comment (syashakash) [reply examples/meta/src/distance/canberra.sg]: Yes, I will. 
First I would add a commit to this  PR with removing the cookbook page and then add a commit for `data` revision.

**Comment by karlnapf** [comment]
Cool, the page is removed, but I don't see a PR for the data repo and you will also need to update the data submodule in this PR here

**Comment by karlnapf** [comment]
The PR has merge conflicts, you will need to rebase against the latest develop

**Comment by karlnapf** [comment]
wrong data submodule revision! Check the diff!

**Comment by syashakash** [comment]
Sorry for the late amend. I hope there are still no merge conflicts and the submodule is the one required. Travis still has one error with the jobs exceeding time limit that means there is some problem with the build, like this would require shortening the build, I guess!

**Comment by karlnapf** [comment]
Tests passed, integration test included. Merging, and thanks! :)

**Pull request merged**